### PR TITLE
Scheduler - make it possible to add OTel @WithSpan automatically

### LINKED ISSUE #29647: Auto-create trace/span context via OTEL for simple `@Scheduled` methods
### Description

I would like to have my scheduled job methods traced by default so that I don't have to add `@WithSpan`on each such method. 

PS: Separated from the Quartz scheduler flavor as advised here: https://github.com/quarkusio/quarkus/issues/28552#issuecomment-1332540691

### Implementation ideas

`quarkus.scheduler.tracing.enabled`, in analogy to [quarkus.scheduler.metrics.enabled](https://quarkus.io/guides/scheduler#quarkus-scheduler_quarkus.scheduler.metrics.enabled)?

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #29647

### CODE REVIEW

File: extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java [hunk 1]
```diff
@@ -366,9 +367,32 @@ public AnnotationsTransformerBuildItem metrics(SchedulerConfig config,
                                     scheduledMethod.declaringClass().name(),
                                     scheduledMethod.name());
                         }
-                    }));
+                    })));
+        }
+    }
+
+    @BuildStep
+    public void metrics(SchedulerConfig config,
```
Comment (famod) [inline extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java:375]:
```suggestion
    public void tracing(SchedulerConfig config,
```
:question:

**Comment by famod** [comment]
I'll try to find some time to test this, but looks pretty straight forward. :+1:

**Review by brunobat — APPROVED** [review]
Thanks @mkouba !
This looks good to me

**Review by famod — APPROVED** [review]
Works as intended, tested with the scheduler-quickstart.

Thanks!

**Comment by famod** [comment]
This MacOS failure is not related, merging...

**Pull request merged**